zulipterminal/ui_tools/buttons.py

```diff
@@ -348,6 +348,7 @@ def mark_muted(self) -> None:
     def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
         if is_command_key("TOGGLE_TOPIC", key):
             # Exit topic view
+            self.view.stream_topic_map[self.stream_id] = self.topic_name
```

Comment:
This can be done in  `View` class in`ui.py`
```
   def update_stream_topic_map(self, stream_id: int, topic_name: str) -> None:
        self.stream_topic_map[stream_id] = topic_name
```
and in `buttons.py`
```
 self.view.update_stream_topic_map(self.stream_id,self.topic_name)
```